docs/_guide/02-writing-templates.md

```diff
@@ -286,4 +286,83 @@ html`${cache(data.showDetails
 When lit-html re-renders a template, it only updates the modified portions: it doesn't create or remove any more DOM than it needs to. But when you switch from one template to another, lit-html needs to remove the old DOM and render a new DOM tree. 
 
 The `cache` directive caches the generated DOM for a given binding and input template. In the example above, it would cache the DOM for both the  `summaryView` and `detailView` templates. When you switch from one view to another, lit-html just needs to swap in the cached version of the new view, and update it with the latest data.
+### Rendering nothing
+Very often you may conditionally render something, but in the other case
+you do not want to render anything at all. So, if you are using ternary 
+operators in one case you might return your template and in the other you
+want to return no content lit-html leaves you with a couple options:
+* an empty string (`''`)
+* null
+* undefined
+* nothing sentinel
+#### An empty string
+If you return an empty string somewhere in your template, an empty text node 
+will be rendered, but to the user it will seem like nothing was rendered. 
+
+````js
+${user.isAdmin
+      ? html`<button>DELETE</button>`
+      : ''
+  }
+````
+This does the trick and works great and efficiently in most cases. It does
+not work with slot fallback content.
+#### null or undefined
+If you return `null` or `undefined` in your template, lit-html will use these as values. 
+This works most of the time, but may lead to unexpected behavior as the DOM itself does not
+handle the two equally:
+
+````js
+document.getElementById('container').innerHTML = null;`
+````
+will produce an empty div while
+````js
+document.getElementById('container').innerHTML = undefined;
+````
+produces a div with the text undefined in it. Also an attribute binding
+with `null` like this
+````js
+html`<input type="text" placeholder=${null}>`
+````
+will set the placeholder attribute to the string null and not an empty value.
+
+The same applies for 
+````js
+html`<input type="text" placeholder=${undefined}>`
+````
+which will set the placeholder attribute to "undefined". If you want to remove the attribute if its value
+is undefined take a look at the [`ifDefined`](#ifdefined) directive.
+#### nothing and the slot fallback content
+`nothing` is a sentinel value provided by lit-html. It really does render nothing, because
+it clears the Part.
+
+Why is this important? Imagine you have a shadow DOM enabled custom element, `shadow-element`, that uses a slot. 
+The template looks like this:
+```js
+html`<slot>Sorry, no content available. I am just fallback content</slot>`;
+``` 
+The slot defines fallback content for when there is no content defined to be put in the slot. 
+So, extending on our first example:
+```js
+import {nothing} from 'lit-html';
+
+
+html`<shadow-element>
```

Comment:
So this example actually _won't_ work as described, because there's whitespace inside the `<shadow-element>`. Those whitespaces turn into text nodes.

So it would have to be:

```
const myTemplate = () =>  html`
      <shadow-element>${user.isAdmin
        ? html`<button>DELETE</button>`
        : nothing
      }</shadow-element>
`;
```

Reply:
Thanks. Good catch.